keycloak-controller/src/main/java/io/enmasse/keycloak/controller/KeycloakRealmParams.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017-2018, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.keycloak.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+
+public class KeycloakRealmParams {
+
+    private static final Logger log = LoggerFactory.getLogger(KeycloakRealmParams.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
+    public static final KeycloakRealmParams NULL_PARAMS = new KeycloakRealmParams(null, null, null, null);
```

Comment:
NULL_PARAMS.getBrowserSecurityHeaders() will NPE.   I also prefer a defensive copy of mutable structures.  So how about changing the constructor to:

this.browserSecurityHeaders = browserSecurityHeaders == null ? Collections.emptyMap() : new HashMap<>(browserSecurityHeaders);